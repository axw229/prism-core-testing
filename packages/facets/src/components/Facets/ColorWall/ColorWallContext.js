// @flow
import React from 'react'
import noop from 'lodash/noop'
import type { Color } from 'src/shared/types/Colors.js.flow'
import { type GridBounds } from './ColorWall.flow'

export type ColorWallA11yContextProps = {
  a11yFocusCell: number[] | void,
  a11yFocusChunk: GridBounds,
  a11yFromKeyboard: boolean,
  a11yFocusOutline: boolean,
  a11yMaintainFocus: boolean
}

export const colorWallA11yContextDefault: ColorWallA11yContextProps = {
  a11yFocusCell: undefined,
  a11yFocusChunk: undefined,
  a11yFromKeyboard: false,
  a11yFocusOutline: false,
  a11yMaintainFocus: false
}

export type ColorWallContextProps = {
  activeColorRouteBuilderRef?: { current: (Color) => void },
  addButtonText?: string,
  autoHeight?: boolean,
  chunkClickable?: boolean,
  chunkMiniMap?: boolean,
  colorDetailPageRoot?: (Color) => string | string,
  colorWallBgColor: string,
  colorWallPageRoot?: (string | Color) => string | string,
  displayAddButton?: boolean,
  displayAddButtonText?: boolean,
  displayDetailsLink?: boolean,
  displayInfoButton?: boolean,
  hiddenSections?: string[],
  inactiveColorRouteBuilderRef?: { current: (Color) => void },
  isChipLocator?: boolean,
  loading: boolean,
  leftHandDisplay: boolean,
  routeType: string,
  swatchMaxSize: number,
  swatchMinSize: number,
  swatchMinSizeZoomed: number,
  swatchSizeZoomed: number,
  updateA11y: Function
}

export const colorWallContextDefault: ColorWallContextProps = {
  activeColorRouteBuilderRef: undefined,
  addButtonText: undefined,
  autoHeight: false,
  chunkClickable: false,
  chunkMiniMap: false,
  colorDetailPageRoot: undefined,
  colorWallBgColor: '#EEEEEE',
  colorWallPageRoot: undefined,
  displayAddButton: false,
  displayAddButtonText: false,
  displayDetailsLink: true,
  displayInfoButton: false,
  hiddenSections: [],
  inactiveColorRouteBuilderRef: undefined,
  isChipLocator: false,
  loading: false,
  leftHandDisplay: false,
  routeType: 'hash',
  swatchMaxSize: 33,
  swatchMinSize: 14,
  swatchMinSizeZoomed: 50,
  swatchSizeZoomed: 50,
  updateA11y: noop,
  viewColorText: false
}

const ColorWallContext = React.createContext<Object>({
  ...colorWallA11yContextDefault,
  ...colorWallContextDefault
})

export default ColorWallContext
