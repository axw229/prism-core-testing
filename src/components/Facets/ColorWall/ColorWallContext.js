import React from 'react'

const ColorWallContext = React.createContext()

export default ColorWallContext
