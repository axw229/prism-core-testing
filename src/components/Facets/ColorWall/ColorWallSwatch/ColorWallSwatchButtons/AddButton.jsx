// @flow
import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import { CLASS_NAMES } from '../shared'

const AddButton = ({ config, onAdd, onClick }: Object) => {
  if (onAdd && config.ColorWall.displayAddButton) {
    return (
      <button /* autoFocus */ onClick={onClick} className={CLASS_NAMES.CONTENT_ADD}>
        <FontAwesomeIcon icon='plus' size='1x' />
      </button>
    )
  }
  return null
}

export default AddButton
