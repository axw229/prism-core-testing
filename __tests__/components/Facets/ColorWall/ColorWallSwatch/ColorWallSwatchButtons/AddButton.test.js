/* eslint-env jest */
import React from 'react'
import AddButton from 'src/components/Facets/ColorWall/ColorWallSwatch/ColorWallSwatchButtons/AddButton'
import ColorWallContext from 'src/components/Facets/ColorWall/ColorWallContext'

const onClickMock = jest.fn()

describe('default AddButton', () => {
  test('should not render', () => expect(mocked(<AddButton />).find(AddButton)).toEqual({}))
})

describe('AddButton wrapped in ColorWallContext with displayAddButton = true', () => {
  let addButton
  beforeAll(() => {
    addButton = mocked(
      <ColorWallContext.Provider value={{ displayAddButton: true }}>
        <AddButton onClick={onClickMock} />
      </ColorWallContext.Provider>
    )
  })

  test('should match snapshot', () => expect(addButton).toMatchSnapshot())

  test('should call onClickMock when button is clicked', () => {
    addButton.simulate('click')
    expect(onClickMock).toHaveBeenCalled()
  })
})
