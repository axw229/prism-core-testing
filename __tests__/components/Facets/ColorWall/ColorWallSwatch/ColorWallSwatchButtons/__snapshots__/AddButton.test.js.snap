// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton should match snapshot 1`] = `
<button
  onClick={[MockFunction]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="plus"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="1x"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</button>
`;

exports[`AddButton with onAdd undefined should match snapshot 1`] = `""`;
